routing.rst

```diff
@@ -290,23 +327,295 @@ with a locale. This can be done by defining a different prefix for each locale
             ;
         };
 
-.. _routing-requirements:
+.. _routing-route-groups:
+
+Route Groups
+~~~~~~~~~~~~
+
+It's common for routes to share some features and options (e.g. all routes
+related to the blog start with ``/blog``; all routes validate the locale with
+the same regular expression; etc.) That's why Symfony includes a feature to
+share the routes configuration.
+
+When defining routes as annotations, add a ``@Route`` annotation to the
+controller class to apply some options to all the routes defined in that class::
+
+    use Symfony\Component\Routing\Annotation\Route;
+
+    /**
+     * @Route("/blog", requirements={"locale": "en|es|fr"}, name="blog_")
+     */
+    class BlogController
+    {
+        /**
+         * @Route("/{_locale}", name="index")
+         */
+        public function index()
+        {
+            // ...
+        }
+
+        /**
+         * @Route("/{_locale}/posts/{slug}", name="post")
+         */
+        public function show(Post $post)
+        {
+            // ...
+        }
+    }
+
+In this example, the route of the ``index()`` action will be called ``blog_index``
+and its URL will be ``/blog/``. The route of the ``show()`` action will be called
+``blog_post`` and its URL will be ``/blog/{_locale}/posts/{slug}``. Both routes
+will also validate that the ``_locale`` parameter matches the regular expression
+defined in the class annotation.
```

Comment:
We could put part/most of this explanation as comments in the code above:

```php
/**
 * @Route("/{_locale}/posts/{slug}", name="post")
 *
 * creates a "/blog/{_locale}/posts/{slug}" route named "blog_post"
 */
```

Reply:
I'm not fan of this because annotations and "normal comments" in the same PHPdoc always look a bit confusing, even for Symfony experts. But I've changed this example a bit to merge annotations and other formats.